src/plugins/content/itemScrolling/itemScrolling.js

```diff
@@ -37,135 +40,203 @@ export default pluginFactory({
-                const overflowCssProp = isVerticalWritingMode ? 'overflow-x' : 'overflow-y';
-
-                if ($item.length && isScrollable) {
-                    $item.data('scrollable', true);
-                    $item.css({ [overflowCssProp]: 'scroll' });
                     if (containerParent.length > 0) {
-                        $item.css('max-block-size', `${containerBlockSize * (selectedBlockSize * 0.01)}px`);
+                        $block.css(maxSizeCssProp, `${containerBlockSize * (selectedBlockSize * 0.01)}px`);
                     } else {
-                        $item.css('max-block-size', `${contentBlockSize * (selectedBlockSize * 0.01)}px`);
+                        const maxSize = contentBlockSize * (selectedBlockSize * 0.01);
+                        if (maxSize > minimalAcceptableSize) {
+                            $block.css(maxSizeCssProp, `${maxSize}px`);
+                        } else {
+                            // contentBlockSize could turn out to be negative or very small because of
+                            //  'getExtraGridRowBlockSize' [other grid-row's content is unexpectedly long] or 'getSpaceAroundQtiContent';
+                            // then we show block with natural size (no scrollbars);
+                            // but for block with different writing-mode need to define *some* size.
+                            if (isBlockVerticalWriting !== isItemVerticalWriting) {
+                                $block.css(maxSizeCssProp, `${innerItemSize / 2}px`);
+                            }
+                        }
+                    }
+
+                    if (isBlockVerticalWriting !== isItemVerticalWriting) {
+                        $block.css('block-size', '100%');
                     }
                 }
             });
         }
-        // depending on the context (item preview, new/old test runner...) available height is computed differently
-        function getItemRunnerBlockSize(isVerticalWritingMode) {
-            var $testRunnerSections = $('.test-runner-sections');
-
-            // exists only in the new test runner
-            if ($testRunnerSections.length) {
-                const rect = $testRunnerSections.get(0).getBoundingClientRect();
-                return isVerticalWritingMode ? rect.width : rect.height;
-            }
-            // otherwise, we assume that we are in an iframe with all space available (= item preview, old test runner)
-            return isVerticalWritingMode ? $(window).width() : $(window).height();
+
+        // may be useful in custom item css,
+        // if scrollable block size must be defined more accurately than what Authoring Editor proposes
+        function defineItemSizeVariable(innerItemSize) {
+            $root
+                .find('.qti-itemBody')
+                .get(0)
+                .style.setProperty('--item-container-inner-block-size', `${innerItemSize}px`);
```

Comment:
Does it make any sense to propose this? Or we are just making things more difficult to support?

The idea is to make it easier for customer to control block size with custom css - to use `%` from item size. Since values defined in Authoring "Use scrolling" dropdown may not be enough. 

Though, for these complex layouts to look good, I guess everything should be defined in `px` for specific media sizes, not in `%`:
```
@media (iPad Air) {
   .grid-row:first-child .colrow:last-child .qti-choiceInteraction { 
      /* !important is to override "style" attr */
      /* 300px defined empirically from size of all content in this grid-row  */
       max-width: 300px !important; 
   }
}
```